pallets/gear/src/lib.rs

```diff
@@ -494,42 +503,78 @@ pub mod pallet {
         }
 
         #[cfg(not(test))]
-        pub fn get_gas_spent(
+        pub fn calculate_gas_info(
             source: H256,
             kind: HandleKind,
             payload: Vec<u8>,
             value: u128,
-        ) -> Result<u64, Vec<u8>> {
-            Self::get_gas_spent_impl(source, kind, payload, value)
+        ) -> Result<GasInfo, Vec<u8>> {
+            let initial_gas = <T as pallet_gas::Config>::BlockGasLimit::get();
+            let GasInfo { spent, to_send, .. } = Self::calculate_gas_info_impl(
+                source,
+                kind.clone(),
+                initial_gas,
+                payload.clone(),
+                value,
+            )?;
+
+            // TODO: adding `to_send` until #642 implemented
+            Self::calculate_gas_info_impl(source, kind, spent + to_send, payload, value).map(
```

Comment:
How will we distinguish the first call error from the second one?

Reply:
Frankly speaking I don't think we need to distinguish them: any error means that required gas cannot be determined

Reply:
Ok, let's collect feedback from users.